### ISSUE TITLE
CA2235 false positive when targeting .NET Standard or .NET Core.

### ISSUE THREAD

**Issue by jvandertil** [dotnet/roslyn-analyzers#1757]
#### Analyzer package
[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version
v2.6.1 and v2.6.2-beta1

#### Diagnostic ID
[CA2235](https://docs.microsoft.com/nl-nl/visualstudio/code-quality/ca2235-mark-all-non-serializable-fields)

#### Repro steps
1. Create a project targeting netstandard2.0 or netcoreapp2.0
2. Create a custom enum type and use this on a custom exception type.
```cs
public enum CustomEnum : int
{
    Success = 0,
    SomethingElse = 1,
}

[Serializable]
public class TestException : Exception
{
    public CustomEnum Error { get; }

    // Constructors ommitted.
}
```
3. Build & Run Analyzer

#### Expected behavior
No CA2235 warning is raised on property Error in TestException as the enum should be serializable by default.

#### Actual behavior
A CA2235 warning is raised on property Error in TestException.

Adding the [Serializable] attribute to CustomEnum does not fix the error either.

The analyzer works fine when targeting full framework (tested with net46).

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/1712
